bson/bsoncodec/bsoncodec.go

```diff
@@ -118,11 +119,34 @@ type EncodeContext struct {
 type DecodeContext struct {
 	*Registry
 	Truncate bool
+
 	// Ancestor is the type of a containing document. This is mainly used to determine what type
 	// should be used when decoding an embedded document into an empty interface. For example, if
 	// Ancestor is a bson.M, BSON embedded document values being decoded into an empty interface
 	// will be decoded into a bson.M.
+	//
+	// Deprecated: Use DefaultDocumentType instead.
 	Ancestor reflect.Type
+
+	// defaultDocumentType specifies the Go type to decode top-level and nested BSON documents into. In particular, the
+	// usage for this field is restricted to data typed as "interface{}" or "map[string]interface{}". If DocumentType is
+	// set to a type that a BSON document cannot be unmarshaled into (e.g. "string"), unmarshalling will result in an
+	// error. DocumentType overrides the Ancestor field.
+	defaultDocumentType reflect.Type
+}
+
+// DefaultDocumentM will set the defaultDocumentType as primitive.M, to be used when decoding "interface{}" and
+// "map[string]interface{}".
+func (dc *DecodeContext) DefaultDocumentM() error {
+	dc.defaultDocumentType = reflect.TypeOf(primitive.M{})
+	return nil
+}
+
+// DefaultDocumentD will set the defaultDocumentType as primitive.D, to be used when decoding "interface{}" and
+// "map[string]interface{}".
```

Comment:
Optional: Consider using the same doc comment as on `Decoder.DefaultDocumentD` which is more clear and doesn't require knowing about the unexported field `defaultDocumentType`.